src/utils/rateLimiter.util.ts

```diff
@@ -0,0 +1,27 @@
+import { RedisService } from '@liaoliaots/nestjs-redis';
+import { Injectable, Logger } from '@nestjs/common';
+
+@Injectable()
+export class RateLimiterService {
+  private readonly logger = new Logger(RateLimiterService.name);
+
+  constructor(private readonly redisService: RedisService) {}
+
+  async isRateLimited(senderId: string, repoId: string): Promise<boolean> {
+    const client = this.redisService.getClient();
+    const key = `rate_limit:${senderId}:${repoId}`;
+    const currentCount = await client.get(key);
+
+    if (currentCount && parseInt(currentCount) >= 5) {
+      return true;
+    }
+
+    await client
+      .multi()
+      .incr(key)
+      .expire(key, 60) // TTL of 60 seconds
+      .exec();
+
+    return false;
```

Comment:
PullProofAI

It would be beneficial to wrap the Redis multi commands in a try-catch block to handle any potential errors that might occur during the execution of these commands.
+     try {
+         await client
+             .multi()
+             .incr(key)
+             .expire(key, 60) // TTL of 60 seconds
+             .exec();
+     } catch (error) {
+         this.logger.error(`Error executing Redis multi commands: ${error.message}`);
+         // Handle the error accordingly
+     }
```


<!-- This is an auto-generated comment by PullProofAI -->